src/tools/wasm2c-wrapper.h

```diff
@@ -25,6 +25,27 @@
 
 namespace wasm {
 
+// Mangle a name in (hopefully) exactly the same way wasm2c does.
+static std::string wasm2cMangle(Name name) {
+  const char escapePrefix = 'Z';
+  std::string mangled;
+  const char* original = name.str;
+  unsigned char c;
+  while ((c = *original++)) {
+    if ((isalnum(c) && c != escapePrefix) || c == '_') {
+      // This character is ok to emit as it is.
+      mangled += c;
+    } else {
+      // This must be escaped, as prefix + hex character code.
+      mangled += escapePrefix;
+      std::stringstream ss;
+      ss << std::hex << std::uppercase << unsigned(c);
+      mangled += ss.str();
+    }
+  }
+  return std::string("(*Z_") + mangled + "Z_";
```

Comment:
I compared this code with wasm2c's [`MangleName`](https://github.com/WebAssembly/wabt/blob/1eb5e2c5b7409a64c6e610fb74df3e5b61394cfc/src/c-writer.cc#L408-L425) method and

Reply:
Thanks, I think you're right, not sure how this worked in my local testing... fixed.

Reply:
Oh, actually the code worked before, but was confusing, sorry about that. I refactored it now so it's hopefully clearer.

What was unclear was that the "mangle" function only mangled part of the name, leaving the params etc. for later. There is another `Z_` before them. I moved all the mangling logic into a single place now